src/Grok.Net.Tests/UnitTests.cs

```diff
@@ -73,29 +78,47 @@ public void TimePatternTest()
         public void EmailPatternTest()
         {
             Grok act = new Grok("%{EMAILADDRESS:email}:%{GREEDYDATA:comment}");
-            string logs = @"Bob.Davis@microsoft.com:Free as in Free Beer";
+            string emailAddress ="Bob.Davis@microsoft.com";
+            string comment = "Free as in Free Beer";
+            string logs = emailAddress + ":" + comment;
             
             GrokResult grokResult = act.Parse(logs);
            
-            Assert.Equal("Bob.Davis@microsoft.com", grokResult[0].Value);
-            Assert.Equal("Free as in Free Beer", grokResult[1].Value);
+            Assert.Equal(emailAddress, grokResult[0].Value);
+            Assert.Equal(comment, grokResult[1].Value);
             Assert.Equal("email", grokResult[0].Key);
             Assert.Equal("comment", grokResult[1].Key);
         }            
 
-        [Fact]
-        public void IPv4PatternTest()
+        [Theory]
+        [MemberData(nameof(GetIpv4PatternTestData))]
+        public void IPv4PatternTest(string grokPattern, string logs, string ipAddress, string comment, int grokElementOffset)
         {
-            Grok act = new Grok("%{IPV4:IP}:%{GREEDYDATA:comment}");
-            string logs = @"172.26.34.32:Free as in Free Beer
-                10.0.12.17:In for the win";
+            Grok act = new Grok(grokPattern);
             
             GrokResult grokResult = act.Parse(logs);
            
-            Assert.Equal("172.26.34.32", grokResult[0].Value);
-            Assert.Equal("Free as in Free Beer", grokResult[1].Value);
-            Assert.Equal("10.0.12.17", grokResult[2].Value);
-            Assert.Equal("In for the win", grokResult[3].Value);            
+            Assert.Equal(ipAddress, grokResult[grokElementOffset].Value);
+            Assert.Equal(comment, grokResult[grokElementOffset+1].Value);
         }  
+        public static IEnumerable<object[]> GetIpv4PatternTestData()
+        { 
+            string grokPattern="%{IPV4:IP}:%{GREEDYDATA:comment}";
```

Comment:
Please follow my comment from previous review. I've asked about define constants, not variables! Because you need strongly define the value that can't be changed

Reply:
I think that's overkill but your repo, your rules.

Reply:
Could you please explain why it is overkill?